pkg/envoy/ads/stream.go

```diff
@@ -141,42 +148,72 @@ func (s *Server) StreamAggregatedResources(server xds_discovery.AggregatedDiscov
 				proxy.GetCertificateSerialNumber(),
 				proxy.GetPodUID())
 
-			proxy.SetLastAppliedVersion(typeURL, ackVersion)
+			proxy.SetLastAppliedVersion(typeURL, requestVersion)
 
-			// In the DiscoveryRequest we have a VersionInfo field.
-			// When this is smaller or equal to what we last sent to this proxy - it is
-			// interpreted as an acknowledgement of a previously sent request.
-			// Such DiscoveryRequest requires no further action.
-			if ackVersion > 0 && ackVersion <= proxy.GetLastSentVersion(typeURL) {
-				log.Debug().Msgf("Skipping request of type %s from Envoy on Pod with UID=%s for resources (%v),  VersionInfo (%d) <= last sent VersionInfo (%d); ACK",
-					typeURL, proxy.GetPodUID(), discoveryRequest.ResourceNames, ackVersion, proxy.GetLastSentVersion(typeURL))
-				continue
+			requestNonce = discoveryRequest.ResponseNonce
+			// Handle first request on stream, should always reply to empty nonce
+			if requestNonce == "" {
+				// Wildcard mode for LDS/CDS can only happen in this context
+				log.Info().Msgf("Proxy %s: empty nonce, allow TypeURL %s request through",
+					proxy.GetCertificateSerialNumber(), typeURL.Short())
+				goto update
```

Comment:
We could add a comment here to say what the purpose is - something like "Skip X, Y, Z etc."